java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -165,6 +165,9 @@ private predicate inputStreamWrapper(Constructor c, int argi) {
 /** An object construction that preserves the data flow status of any of its arguments. */
 private predicate constructorStep(Expr tracked, ConstructorCall sink) {
   exists(int argi | sink.getArgument(argi) = tracked |
+    summaryStep(any(DataFlow::Node n | n.asExpr() = tracked),
+      any(DataFlow::Node n | n.asExpr() = sink), "taint", "Argument(" + argi + ")", "ReturnValue")
+    or
```

Comment:
This is super ugly.

Reply:
There should be no need for this if you change the constructor steps slightly: Instead of `ReturnValue`, which only works for `new` expressions, you can use `Argument[-1]`, which is the value of `this` for the constructor. This amounts to the same thing for `new` expressions, but also works for other constructor calls such as `this(..)` and `super(..)`.

Reply:
Thanks. Did you mean it like [this](https://github.com/github/codeql/pull/5329/commits/d8cb588629db618772636a509700785ec407007d)?